analytics_data_api/v0/tests/views/test_learners.py

```diff
@@ -185,3 +189,227 @@ def test_bad_course_id(self):
             u"developer_message": u"Course id/key malformed-course-id malformed."
         }
         self.assertDictEqual(json.loads(response.content), expected)
+
+
+@ddt.ddt
+class LearnerListTests(LearnerAPITestMixin, TestCaseWithAuthentication):
+    """Tests for the learner list endpoint."""
+    def setUp(self):
+        super(LearnerListTests, self).setUp()
+        self.course_id = 'edX/DemoX/Demo_Course'
+
+    def _get(self, course_id, **query_params):
+        """Helper to send a GET request to the API."""
+        query_params['course_id'] = course_id
+        return self.authenticated_get('/api/v0/learners/', query_params)
+
+    def assert_learners_returned(self, response, expected_learners):
+        """
+        Verify that the learners in the response match the expected
+        learners, in order.  Each learner in `expected_learners` is a
+        dictionary subset of the expected returned representation.  If
+        `expected_learners` is None, assert that no learners were
+        returned.
+        """
+        self.assertEqual(response.status_code, 200)
+        payload = json.loads(response.content)
+        returned_learners = payload['results']
+        if expected_learners is None:
+            self.assertEqual(returned_learners, list())
+        else:
+            for expected_learner, returned_learner in zip(expected_learners, returned_learners):
```

Comment:
It may be worth it check that the size of the returned and expected lists are the same before the loop.